File: sol/1282/1282.cpp
#include <unordered_map>

struct Solution {
    vector<vector<int>> groupThePeople(vector<int>& groupSizes) {
        unordered_map<int, vector<int>> d;
        vector<vector<int>> res;
        for (int i = 0; i < groupSizes.size(); ++i){
            d[groupSizes[i]].push_back(i);
            if (d[groupSizes[i]].size() == groupSizes[i]){
                res.push_back(d[groupSizes[i]]);
                d[groupSizes[i]].clear();
            }
        }
        return res;
    }
};

File: sol/1282/1282.py
from collections import defaultdict

class Solution:
    def groupThePeople(self, groupSizes: List[int]) -> List[List[int]]:
        d = defaultdict(list)
        res = []
        for i, x in enumerate(groupSizes):
            d[x].append(i)
            if len(d[x]) == x:
                res.append(d[x].copy())
                d[x].clear()
        return res
